src/components/Badge/Badge.driver.ts

```diff
@@ -3,28 +3,37 @@ import { badgeDriverFactory as coreBadgeDriverFactory, BadgeDriver as CoreBadgeD
 import { uiTextDriverFactory, UITextDriver } from '../StylableUIText/UIText.driver';
 import { StylableDOMUtil } from '@stylable/dom-test-kit';
 import style from './Badge.st.css';
-import { Type, Skin } from './constants';
+import { Type, Skin, Size } from './constants';
 
 export interface BadgeDriver extends CoreBadgeDriver {
   getType: () => Type,
   getSkin: () => Skin,
+  getSize: () => Size,
+  getUppercase: () => boolean,
+  getHasClickCursor: () => boolean,
   getUIText: () => UITextDriver,
   getPrefixIcon: () => Element | null,
-  getSuffixIcon: () => Element | null
+  getSuffixIcon: () => Element | null,
+  click: () => void
 }
 
+
 export const badgeDriverFactory = (factoryParams: ComponentFactory): BadgeDriver => {
   const coreBadgeDriver = coreBadgeDriverFactory(factoryParams);
-  const { element } = factoryParams;
+  const { element, eventTrigger } = factoryParams;
   const stylableDOMUtil = new StylableDOMUtil(style, element);
   const uiTextDriver = uiTextDriverFactory({ ...factoryParams, element: stylableDOMUtil.select('.text') });
 
   return {
     ...coreBadgeDriver,
     getType: () => stylableDOMUtil.getStyleState(element, 'type') as Type,
     getSkin: () => stylableDOMUtil.getStyleState(element, 'skin') as Skin,
+    getSize: () => stylableDOMUtil.getStyleState(element, 'size') as Size,
+    getUppercase: () => stylableDOMUtil.getStyleState(element, 'uppercase') === 'true',
+    getHasClickCursor: () => stylableDOMUtil.getStyleState(element, 'hasOnClickHandler') === 'true',
```

Comment:
visual changes should be done in the e2e/ browser tests. so let's remove it and the test for it and add an e2e scenario for it